modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/CachePartitionLossWithPersistenceTest.java

```diff
@@ -162,6 +175,142 @@ public void testConsistencyAfterResettingLostPartitions_3() throws Exception {
         doTestConsistencyAfterResettingLostPartitions(2, false);
     }
 
+    /**
+     * Tests data recovery after restarting the cluster with disabled WAL on rebalancing.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testConsistencyAfterClusterRestartWithDisabledWalOnRebalancing() throws Exception {
+        doTestConsistencyAfterClusterRestartWithDisabledWalOnRebalancing(false);
+    }
+
+    /**
+     * Tests data recovery after restarting the cluster in the case where one node of the cluster was stopped in the middle of checkpoint.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testConsistencyAfterClusterRestart() throws Exception {
+        doTestConsistencyAfterClusterRestartWithDisabledWalOnRebalancing(true);
+    }
+
+    /**
+     * Tests data recovery after restarting the cluster with disabled WAL on rebalancing.
+     *
+     * @throws Exception If failed.
+     */
+    public void doTestConsistencyAfterClusterRestartWithDisabledWalOnRebalancing(boolean stopInTheMiddleOfCheckpoint) throws Exception {
+        lossPlc = READ_WRITE_SAFE;
+
+        IgniteEx ig0 = startGrids(2);
+
+        ig0.cluster().state(ACTIVE);
+
+        IgniteCache<Integer, Integer> cache = ig0.cache(DEFAULT_CACHE_NAME);
+
+        // Initial load.
+        for (int k = 0; k < 2500; k++)
+            cache.put(k, k);
+
+        // Stop the second node and update data in the cache to trigger rebalancing when the node will be restarted.
+        stopGrid(1);
+
+        int owningPartId = 12;
+
+        // Rewrite data to trigger further rebalance.
+        for (int k = 2500; k < 5000; k++) {
+            if (!stopInTheMiddleOfCheckpoint) {
```

Comment:
I would not use a negative over a whole condition in a clause with else.

Reply:
Ok. Fixed.